core/frontend/src/components/app/ImagePicker.vue

```diff
@@ -0,0 +1,255 @@
+              @dragenter.prevent
+              @drop.prevent="onDrop"
+            >
+              <v-card-text>
+                Drag and drop an image file here to upload
+              </v-card-text>
+              <input
+                id="file-input"
+                ref="fileInput"
+                aria-label="File browser"
+                type="file"
+                accept="image/*"
+                style="display:none"
+                @change="onFileInputChange"
+              />
+              <v-btn color="primary" @click="onFilePickerClick">
+                Click to upload an image or
+              </v-btn>
+            </v-card>
+          </v-col>
+        </v-row>
+        <v-alert v-if="uploadError" color="red lighten-2">
+          {{ uploadError }}
+        </v-alert>
+      </v-card>
+    </v-dialog>
+  </div>
+</template>
+
+<script lang="ts">
+import axios from 'axios'
+import Vue from 'vue'
+
+import back_axios from '@/utils/api'
+
+import SpinningLogo from '../common/SpinningLogo.vue'
+
+interface FileEntry {
+  name: string;
+  type: string;
+  mtime: string;
+}
+
+export default Vue.extend({
+  name: 'ImagePicker',
+  components: {
+    SpinningLogo,
+  },
+  props: {
+    directory: {
+      type: String,
+      default: '/userdata/images/',
+      required: false,
+    },
+    image: {
+      type: String,
+      default: '/userdata/images/',
+      required: false,
+    },
+  },
+  data() {
+    return {
+      dialog: false,
+      selectedIndex: null as number | null,
+      images: [] as string[],
+      error: null as string | null,
+      loading: true,
+      uploadError: null as string | null,
+    }
+  },
+  computed: {
+  },
+  mounted() {
+    this.loadImages()
+  },
+  methods: {
+    selectImage(index: number) {
+      this.selectedIndex = index
+      this.$emit('image-selected', `${this.directory}/${this.images[index]}`)
+      this.dialog = false
+    },
+    openDialog() {
+      this.dialog = true
+    },
+    loadImages() {
+      back_axios({
+        method: 'get',
+        url: this.directory,
+      }).then((response) => {
+        this.images = response.data.filter(
+          (string: FileEntry) => string.type === 'file',
+        ).map((file: FileEntry) => file.name)
+        this.loading = false
+      }).catch((error) => {
+        this.error = error
+      })
+    },
+    async deleteImage(index: number) {
+      try {
+        await back_axios({
+          method: 'delete',
+          url: `upload/${this.directory}/${this.images[index]}`,
+        })
+        this.loadImages()
+      } catch (error) {
+        console.error('Error deleting file:', error)
+      }
+    },
+    onFileInputChange(event: Event) {
+      const target = event.target as HTMLInputElement
+      if (target.files && target.files.length > 0) {
+        const file = target.files[0]
+        const fileName = encodeURIComponent(file.name)
+        this.uploadFile(file, `${this.directory}/${fileName}`)
+      }
```

Comment:
```suggestion
      const file = (event.target as HTMLInputElement).files?.[0];
      if (!file) return;
      const fileName = encodeURIComponent(file.name);
      this.uploadFile(file, `${this.directory}/${fileName}`);
```

Reply:
our linter things the parenthesis on ` (event.target as HTMLInputElement)` are redundant 🤦